.github/workflows/main_ci.yml

```diff
@@ -9,18 +9,23 @@ on:
       - main
 
 env:
+  CMAKE_TOOLCHAIN_FILE: ${{ github.workspace }}/vcpkg/scripts/buildsystems/vcpkg.cmake
   CMAKE_BUILD_PARALLEL_LEVEL: 3
   CTEST_OUTPUT_ON_FAILURE: 1
-  CMAKE_BUILD_DIR: ${{ github.workspace }}/build
-  CMAKE_BUILD_OPENSSL3_DIR: ${{ github.workspace }}/build_openssl3
-  CMAKE_TEST_OPENSSL3_DIR: ${{ github.workspace }}/build_openssl3/test
-  CMAKE_TEST_DIR: ${{ github.workspace }}/build/test
+
+  BUILD_DIR: ${{ github.workspace }}/build
+  BUILD_DIR_OPENSSL3: ${{ github.workspace }}/build_openssl3
+  BUILD_DIR_BORINGSSL: ${{ github.workspace }}/build_boringssl
+
   VCPKG_BINARY_SOURCES: files,${{ github.workspace }}/build/cache,readwrite
-  VCPKG_TOOLCHAIN_FILE: ${{ github.workspace }}/vcpkg/scripts/buildsystems/vcpkg.cmake
   VCPKG_REPO: ${{ github.workspace }}/vcpkg
   CACHE_VERSION: v01
   CACHE_NAME: vcpkg
 
+  OPENSSL_11_VCPKG: alternatives/openssl_1.1
+  OPENSSL_3_VCPKG: alternatives/openssl_3
+  BORINGSSL_VCPKG: alternatives/boringssl
```

Comment:
I find myself getting confused about which vcpkg attribute we might be talking about.  I would recommend renaming this to `*_VCPKG_MANIFEST` to match the cmake option.